packages/playground/src/utils/validators.ts

```diff
@@ -777,3 +777,20 @@ export async function isAvailableName(grid: GridClient, name: string) {
     return { message: "Name is already taken." };
   }
 }
+
+/**
+ * Validates an SMTP input string.
+ *
+ * Checks if the input string is a valid email address and does not contain special characters.
+ *
+ * @param {string} input - The input string to validate.
+ * @returns {{ message: string }} - An object with an error message if the input is invalid, or undefined if the input is valid.
+ */
+
+export function isValidSmtp(input: string) {
+  const emailValidation = isEmail("Please provide a valid email address.")(input);
+  const username = /[!@#$%^&*()\s_+\-={}:<>?,./]/.test(input);
+  if (username && emailValidation) {
```

Comment:
I don't understand this condition, does that mean if it's email and a username (doesn't have special characters) then it's true? 

shouldn't it be like that

```
const isValidEmail = isEmail("Please provide a valid email address.")(input);
const hasSpecialCharacters = /[!@#$%^&*()\s_+\-={}:<>?,./]/.test(input);

if (hasSpecialCharacters && isValidEmail) {
      return { message: "Please provide a valid username or email" };
}
```

Which I still don't understand.

Reply:
I meant here it could accept both and validate if it's not a valid email or username with a special character.